index.html

```diff
@@ -0,0 +1,513 @@
+          }
+        }
+        if (!Array.isArray(inner)) {
+          inner = [inner];
+        }
+        for (var k = 0; k < inner.length; k++) {
+          if (inner[k].tagName) {
+            el.appendChild(inner[k]);
+          } else {
+            el.appendChild(document.createTextNode(inner[k]));
+          }
+        }
+        return el;
+    }
+
+    var container = createElement('div', {'class':'container'});
+    document.body.appendChild(container);
+
+    var tButton = createElement(
+      'button', 
+      {
+        'class': 'button',
+        'onClick': 'resetParticles()'
+      },
+      'Reset Particles'
+    );
+    var dButton = createElement('div',{}, tButton);
+
+    var option4k = createElement('option',{'value': '4'}, '4K Particles');
+    var option8k = createElement('option',{'value': '8', 'selected': 'selected'}, '8K Particles');
+    var option16k = createElement('option',{'value': '16'}, '16K Particles');
+    var option32k = createElement('option',{'value': '32'}, '32K Particles');
+    var tSelectP = createElement(
+      'select',
+      {'id': 'selP',
+       'onchange': 'restart(this.value)'},
+      [option4k, option8k, option16k, option32k]
+    );
+    var dSelectP = createElement('div',{}, tSelectP);
+
+    var optionDown = createElement('option',{'value': 'down'}, 'Gravity Down');
+    var optionUp = createElement('option',{'value': 'up'}, 'Gravity Up');
+    var optionLeft = createElement('option',{'value': 'left'}, 'Gravity Left');
+    var optionRight = createElement('option',{'value': 'right'}, 'Gravity Right');
+    var tSelectG = createElement(
+      'select',
+      {'id': 'selG'},
+      [optionDown, optionUp, optionLeft, optionRight]
+    );
+    var dSelectG = createElement('div',{}, tSelectG);
+
+    var radioSimple = createElement('input', 
+      {'type': 'radio',
+       'name': 'mode',
+       'id': 'radio1',
+       'value': 'simple',
+       'onchange': 'changeMode(this)',
+       'checked': 'checked'});
+    var lableSimple = createElement('label', {}, 'Simple N^2');
+    var dRadioSimple = createElement('div', {}, [radioSimple, lableSimple]);
+
+    var radioShared = createElement('input',
+      {'type': 'radio',
+       'name': 'mode',
+       'id': 'radio1',
+       'onchange': 'changeMode(this)',
+       'value': 'shared'});
+    var lableShared = createElement('label', {}, 'Shared Memory N^2');
+    var dRadioShared = createElement('div', {}, [radioShared, lableShared]);
+
+    container.appendChild(dButton);
+    container.appendChild(dSelectP);
+    container.appendChild(dSelectG);
+    container.appendChild(dRadioSimple);
+    container.appendChild(dRadioShared);
+}
+
+function changeMode(o) {
+  if (o.value == 'simple') {
+    g_currentDen = g_programs[CS_DEN_SIM];
+    g_currentFor = g_programs[CS_FOR_SIM];
+  } else {
+    g_currentDen = g_programs[CS_DEN_SHA];
+    g_currentFor = g_programs[CS_FOR_SHA];
+  }
+}
+
+function resetParticles() {
```

Comment:
Add a `destroy` function to delete all buffer resources before initBuffers

Reply:
Done, Please have a review